From d8334dbd3f9aaac696054b808b9bc1cf5f3bda57 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Wed, 11 Jan 2023 05:55:51 +0700
Subject: [PATCH] Fixes for Macports

1. Do not build internal LAPACK, since linking to it fails: https://github.com/Dsplib/libdspl-2.0/issues/5
2. Use the right compiler.
3. Add missing header.
4. Do not link to libquadmath on PPC, it is not there.

diff --git a/dspl/Makefile b/dspl/Makefile
index 2113580..e71b119 100644
--- dspl/Makefile
+++ dspl/Makefile
@@ -16,8 +16,8 @@ all: $(RELEASE_DIR)/$(LIB_NAME)\
      $(RELEASE_DIR)/dspl.h
 
 #Build libdspl.dll or libdspl.so
-$(RELEASE_DIR)/$(LIB_NAME): $(DSPL_OBJ_FILES)  $(BLAS_LIB_NAME) $(LAPACK_DOUBLE_LIB_NAME) $(LAPACK_COMPLEX_LIB_NAME)
-	$(CC) -shared -o $(RELEASE_DIR)/$(LIB_NAME)  $(DSPL_OBJ_FILES) -lm  -L$(LAPACK_RELEASE_DIR) -llapack_complex -llapack_double -L$(BLAS_RELEASE_DIR) -lblas -lgfortran -lquadmath
+$(RELEASE_DIR)/$(LIB_NAME): $(DSPL_OBJ_FILES)
+	$(CC) -shared -o $(RELEASE_DIR)/$(LIB_NAME) $(DSPL_OBJ_FILES) -lm -L@PREFIX@/lib -llapack -lblas $(EXTRAFLIBS)
 
 #Compile libdspl obj files from c sources
 $(DSPL_OBJ_DIR)/%.o:$(DSPL_SRC_DIR)/%.c
@@ -35,18 +35,6 @@ $(PERFORMANCE_BIN_DIR)/$(LIB_NAME):$(RELEASE_DIR)/$(LIB_NAME)
 $(VERIFICATION_BIN_DIR)/$(LIB_NAME):$(RELEASE_DIR)/$(LIB_NAME)
 	cp $(RELEASE_DIR)/$(LIB_NAME) $(VERIFICATION_BIN_DIR)/$(LIB_NAME)
 
-#make BLAS library
-$(BLAS_LIB_NAME):
-	$(MAKE) -C blas
-
-#make LAPACK library for the real data
-$(LAPACK_DOUBLE_LIB_NAME):
-	$(MAKE) -C lapack
-
-#make LAPACK library for the complex data
-$(LAPACK_COMPLEX_LIB_NAME):
-	$(MAKE) -C lapack
-
 #Copy dspl.c source to the release folder 
 $(RELEASE_DIR)/dspl.c:
 	cp $(INC_DIR)/dspl.c $(RELEASE_DIR)/dspl.c
diff --git a/dspl/src/gnuplot.c b/dspl/src/gnuplot.c
index e131269..82acb27 100644
--- dspl/src/gnuplot.c
+++ dspl/src/gnuplot.c
@@ -19,6 +19,7 @@
 */
 #include <stdio.h>
 #include <unistd.h>
+#include <string.h>
 #include "dspl.h"
 
 #define GNUPLOT_NO     1
@@ -489,7 +490,3 @@ int DSPL_API gnuplot_open(void** hplot)
         return ERROR_GNUPLOT_CREATE;
     return RES_OK;
 }
-
-
-
-
diff --git a/make.inc b/make.inc
index 6f4cdac..13cc35d 100644
--- make.inc
+++ make.inc
@@ -1,28 +1,11 @@
-CC       = gcc
-FORTRAN  = g++
+CC       = @CC@
+FORTRAN  = @FC@
 AR       = ar
 
 # DSPL source and obj file path
 DSPL_SRC_DIR = src
 DSPL_OBJ_DIR = obj
 
-# BLAS source path
-BLAS_LIB_DIR = blas
-BLAS_SRC_DIR = $(BLAS_LIB_DIR)/src
-BLAS_RELEASE_DIR = $(BLAS_LIB_DIR)/bin
-BLAS_LIB_NAME = $(BLAS_RELEASE_DIR)/libblas.a
-
-# LAPACK source path
-LAPACK_LIB_DIR  =   lapack
-LAPACK_RELEASE_DIR = $(LAPACK_LIB_DIR)/bin
-
-LAPACK_DOUBLE_SRC_DIR  = $(LAPACK_LIB_DIR)/src
-LAPACK_DOUBLE_LIB_NAME = $(LAPACK_RELEASE_DIR)/liblapack_double.a
-
-LAPACK_COMPLEX_SRC_DIR  = $(LAPACK_LIB_DIR)/src
-LAPACK_COMPLEX_LIB_NAME = $(LAPACK_RELEASE_DIR)/liblapack_complex.a
-
-
 
 INC_DIR = ../include
 RELEASE_DIR = ../_release
@@ -34,6 +17,7 @@ ifeq ($(OS),Windows_NT)
 	DSPL_LIBNAME = libdspl.dll
 	DEF_OS = WIN_OS
 	LFLAGS = -lm
+	EXTRAFLIBS = -lgfortran -lquadmath
 else
 	UNAME_S := $(shell uname -s)
 	UNAME_P := $(shell uname -p)
@@ -41,10 +25,16 @@ else
 		DSPL_LIBNAME = libdspl.so
 		DEF_OS = LINUX_OS
 		LFLAGS = -lm -ldl
+		EXTRAFLIBS = -L@PREFIX@/lib/libgcc -lgfortran -lquadmath
 	else ifeq ($(UNAME_S),Darwin)
 		DSPL_LIBNAME = libdspl.so
 		DEF_OS = LINUX_OS
 		LFLAGS = -lm -ldl
+		ifeq ($(UNAME_P),powerpc)
+			EXTRAFLIBS = -L@PREFIX@/lib/libgcc -lgfortran
+		else
+			EXTRAFLIBS = -L@PREFIX@/lib/libgcc -lgfortran -lquadmath
+		endif
 	endif
 endif
 
@@ -60,4 +50,4 @@ else
     else ifeq ($(UNAME_S),Darwin)
     	MAKE = make
     endif
-endif
\ No newline at end of file
+endif
